blog/2024-03-06-rag-tool-vercel-gen-ai/index.mdx

```diff
@@ -0,0 +1,443 @@
+This approach not only harnesses the LLM's generative capabilities but also its ability to make contextual decisions on-the-fly, thereby opening up a more dynamic and responsive way to handle knowledge search tasks.
+As a result, you can build targeted KB's and use the power of LLMs to write application-specific queries that retrieve better, more relevant results.
+
+```mermaid
+sequenceDiagram title RAG Tool
+    actor User
+    participant Service
+    participant LLM
+    participant KB
+
+    User->>+Service: What is the meaning of life
+
+    Service->>+LLM: What is the meaning of life
+    LLM-->-Service: search_kb('meaning of life')
+    Note over Service,LLM: LLM Selects Tools
+
+    Service->>+KB: meaning of life
+    KB-->-Service: [42, Coffee]
+    Note over Service,KB: LLM-Directed Retrieval
+
+    Service->>+LLM: What is the meaning of life (context: [42, Coffee])
+    LLM-->>-Service: 42
+    Note over Service,LLM: Augmented Generation
+
+    Service-->>-User: 42
+```
+
+
+To demonstrate this approach, we'll build [a simple chatbot](https://github.com/DewyKB/dewy-ts/tree/main/examples) backed by a custom tool.
+The chatbot will have access to a KB containing product information - any time you ask a question the chatbot determines could be answered using this tool it will query the KB and show both the in-progress query and retrieved results as custom UI elements before responding with the retrieved products as context.
+
+![Screen capture of finished app](screencap.gif)
+
+We'll use Vercel's Generative UI library to render the chat window and Dewy to implement the information storage and retrieval used by our tool. 
+
+## Why Dewy and Vercel?
+
+**Vercel's [Generative UI library](https://sdk.vercel.ai/docs/concepts/ai-rsc)** is designed to simplify the creation of custom LLM "tools" and to render rich, interactive UI elements that are specific to each tool the LLM chooses to use. 
+What this means in practice is a seamless and engaging user experience, where the interface adapts and evolves based on the context of the query and the tools selected by the LLM. 
+
+**[Dewy](https://github.com/DewyKB/dewy)** is an OSS knowledge base designed to streamline the way developers store, organize, and retrieve information. 
+Dewy is a knowledge base with the simplicity and ease-of-use of a document store:  insert your documents and Dewy takes care of preparing them for semantic searching. Dewy's flexibility and ease of use make it an excellent choice for developers aiming to build custom knowledge-driven LLM tools.
+
+## Prerequisites
+
+Before diving into the tutorial, ensure you have the following prerequisites covered:
+
+* Basic knowledge of Typescript and React
+* A NextJS environment setup on your local machine
+* A copy of Dewy running on your local machine (see Dewy's [installation instructions](http://localhost:3001/docs/Getting%20Started/installation) if you need help here).
```

Comment:
This should be `../docs/GettingStarted/installation` so it's relative where we deploy. Also... localhost *probably* won't work for the user...